operator/bucketcontroller/connector.go

```diff
@@ -0,0 +1,119 @@
+package bucketcontroller
+
+import (
+	"context"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"github.com/vshn/provider-exoscale/operator/controllerutil"
+	"github.com/vshn/provider-exoscale/operator/pipelineutil"
+	"net/url"
+	"strings"
+
+	pipeline "github.com/ccremer/go-command-pipeline"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	corev1 "k8s.io/api/core/v1"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+type bucketConnector struct {
+	controllerutil.GenericConnector
+}
+
+type connectContext struct {
+	controllerutil.GenericConnectContext
+	MinioClient *minio.Client
+	EndpointURL string
+}
+
+// Connect implements managed.ExternalConnecter.
+func (c *bucketConnector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	ctx = pipeline.MutableContext(ctx)
+	log := controllerruntime.LoggerFrom(ctx)
+	log.V(1).Info("Connecting resource")
+
+	bucket := fromManaged(mg)
+
+	if isBucketAlreadyDeleted(bucket) {
+		// We do this to prevent after-deletion reconciliations since there is a chance that we might not have the access credentials anymore.
```

Comment:
This sentence isn't true for the exoscale provider though? We're using the admin credentials from the provider config. So I'd remove this comment.

Not doing a reconciliation loop if the bucket doesn't exist still makes sense though.

Reply:
yes, the comment is miss-leading.